protocols/v2/noise-sv2/src/initiator.rs

```diff
@@ -1,3 +1,70 @@
+// # Noise NX Protocol Initiator
+//
+// This module manages the state and operations of the initiator within the Noise NX protocol.
+// Plays a crucial role in the cryptographic handshake, establishing a secure
+// communication channel with the responder by performing key exchanges, deriving shared secrets,
+// and encrypting/decrypting handshake messages.
+//
+// ## Overview
+//
+// The [`Initiator`] is responsible for generating and managing the necessary cryptographic keys,
+// including ephemeral and optional static keys, as well as maintaining the state of the handshake
+// process. It interacts with the responder to derive shared secrets and set up secure cipher
+// states that protect subsequent communication. The struct implements the
+// [`CipherState`] and [`HandshakeOp`] traits to manage the cryptographic state and operations
+// throughout the handshake.
+//
+// The module utilizes cryptographic components such as [`ChaCha20Poly1305`] for encryption,
+// Elligator Swift for key obfuscation, and Secp256k1 for key generation, ensuring that the
+// handshake is both secure and efficient.
+//
+// ## Usage
+//
+// The [`Initiator`] can be instantiated with or without knowledge of the responder's public key.
+// The handshake process is executed in multiple steps, starting with the generation of an
+// ephemeral keypair and encoding the public key, followed by processing messages exchanged with
+// the responder to derive shared secrets and authenticate the responder. Upon successful
+// completion of the handshake, the [`Initiator`] sets up cipher states that are used for encrypted
+// communication with the responder.
+//
+// ### Example
+//
+// This example demonstrates how to initialize an [`Initiator`], perform the first step of the
+// Noise NX handshake, and handle the responder's reply.
+//
+// ```rust
+// use secp256k1::XOnlyPublicKey;
+// use noise_sv2::Initiator;
+// use const_sv2::INITIATOR_EXPECTED_HANDSHAKE_MESSAGE_SIZE;
+//
+// // Optionally, provide the responder's public key
+// let responder_pub_key: Option<XOnlyPublicKey> = None;
+//
+// // Create a new Initiator instance
+// let mut initiator = Initiator::new(responder_pub_key);
+//
+// // Perform the first step of the handshake
+// let encoded_key = initiator.step_0().unwrap();
+// println!("Encoded ephemeral key: {:?}", encoded_key);
+//
+// // Simulate receiving a handshake message from the responder
+// let responder_message = [0u8; INITIATOR_EXPECTED_HANDSHAKE_MESSAGE_SIZE]; // Example message
+//
+// // Process the responder's message and complete the handshake
+// let noise_codec = initiator.step_2(responder_message);
+//
+// println!("Handshake complete, ready for secure communication.");
+// ```
```

Comment:
Should we delete this since we have an example now? Or do we think this adds extra value that the newly added example does not?